pkg/txm/attempt_builder.go

```diff
@@ -42,13 +53,52 @@ func (a *attemptBuilder) NewAttempt(ctx context.Context, lggr logger.Logger, tx
 }
 
 func (a *attemptBuilder) NewBumpAttempt(ctx context.Context, lggr logger.Logger, tx *types.Transaction, previousAttempt types.Attempt) (*types.Attempt, error) {
-	bumpedFee, bumpedFeeLimit, err := a.EvmFeeEstimator.BumpFee(ctx, previousAttempt.Fee, tx.SpecifiedGasLimit, a.priceMaxKey(tx.FromAddress), nil)
+	gasLimit := tx.SpecifiedGasLimit
+	if tx.IsPurgeable {
+		gasLimit = a.emptyTxLimitDefault
+	}
+	bumpedFee, bumpedFeeLimit, err := a.EvmFeeEstimator.BumpFee(ctx, previousAttempt.Fee, gasLimit, a.priceMaxKey(tx.FromAddress), nil)
 	if err != nil {
 		return nil, err
 	}
 	return a.newCustomAttempt(ctx, tx, bumpedFee, bumpedFeeLimit, previousAttempt.Type, lggr)
 }
 
+func (a *attemptBuilder) NewAgnosticBumpAttempt(ctx context.Context, lggr logger.Logger, tx *types.Transaction, dynamic bool) (*types.Attempt, error) {
+	attempt, err := a.NewAttempt(ctx, lggr, tx, dynamic)
+	if err != nil {
+		return nil, err
+	}
+
+	// bump purge attempts
+	if tx.IsPurgeable {
+		// TODO: add better handling
+		for {
+			bumpedAttempt, err := a.NewBumpAttempt(ctx, lggr, tx, *attempt)
+			if err != nil {
+				if errors.Is(err, fees.ErrConnectivity) {
```

Comment:
I'm looking at the error choice. I believe this is due to the code located [here](https://github.com/smartcontractkit/chainlink-evm/blob/da6e4a9838e94e56df48a3dd2daf8429a698742e/pkg/gas/fee_history_estimator.go#L388C2-L392C1)

```
if bumpedMaxPriorityFeePerGas.Cmp(priorityFeeThreshold) > 0 {
    return ..., ErrConnectivity
}
```

It feels slightly counterintuitive to return ErrConnectivity for a value check, where ErrBumpFeeExceedsLimit might seem more natural.

Is the intention to treat priorityFeeThreshold as a sanity guardrail (implying the network/estimator is broken if fees get this high) rather than a budget limit (implying the user just needs to pay more)? If so, adding a small comment explaining why this triggers a 'connectivity' error would be helpful for future readers.

Reply:
This functionality is based on TXMv1. Both [FeeHistory](https://github.com/smartcontractkit/chainlink-evm/blob/da6e4a9838e94e56df48a3dd2daf8429a698742e/pkg/gas/fee_history_estimator.go#L388C2-L392C1) and [BlockHistory](https://github.com/smartcontractkit/chainlink-evm/blob/develop/pkg/gas/block_history_estimator.go#L306) estimators use the 90th percentile as a maximum threshold to prevent indefinite gas bumping.
The above was used in a hacky way to access that percentile without changing multiple interfaces in the gas package during testing. We'll have to update the PR and change this before it gets merged.